runtime/lua/vim/lsp/completion.lua

```diff
@@ -359,7 +359,8 @@ local function request(clients, bufnr, win, callback)
   local request_ids = {} --- @type table<integer, integer>
   local remaining_requests = vim.tbl_count(clients)
 
-  for client_id, client in pairs(clients) do
+  for _, client in pairs(clients) do
+    local client_id = client.id
```

Comment:
This extra variable seem superfluous. You can replace all subsequent usages of `client_id` with `client.id`.

Reply:
It's perfectly fine to use the extra variable. Please no style nitpicking if there isn't a substantial advantage to do it in another way.